lib/spack/spack/test/versions.py

```diff
@@ -630,6 +630,14 @@ def test_version_wrong_idx_type():
         v['0:']
 
 
+@pytest.mark.regression('29170')
+def test_version_range_satisfies_means_nonempty_intersection():
+    x = VersionRange('3.7.0', '3')
+    y = VersionRange('3.6.0', '3.6.0')
+    assert not x.satisfies(y)
+    assert not y.satisfies(x)
```

Comment:
It's not tied to that discussion at all. `3.7.0:3` means `[3.7.0, 4)`; the test you're commenting on verifies that, and fails on `develop`. This is because of a bug on `develop`.